src/google/protobuf/compiler/ruby/ruby_generator.cc

```diff
@@ -323,6 +330,9 @@ bool GenerateFile(const FileDescriptor* file, io::Printer* printer,
   for (int i = 0; i < file->enum_type_count(); i++) {
     GenerateEnumAssignment("", file->enum_type(i), printer);
   }
+  for (int i = 0; i < file->extension_count(); i++) {
```

Comment:
This handles the case of top-level extensions, but we also need to handle the case of extensions nested in other messages.

Maybe add a test for this case, and then change the code generator accordingly?

Reply:
Added, plus tests for MessageSet.